### PR TITLE
Add User Registration Page

### PR DESCRIPTION
**Pull request by jazwu**
## Description
add register function in views.py to handle POST requests from registration.html
add registration path to urls.py

## How did you test the changes?
I filled in the form on the registration page, and I clicked the Register button to submit
image
It would redirect to index page
image

## Testing Screenshots
Check the result in the database. User data (id=6) is successfully created.
image

### CODE REVIEW

File: healthScore/healthConfig/settings.py [hunk 1]
```diff
@@ -82,7 +82,7 @@
         'ENGINE': 'django.db.backends.mysql',
         'NAME': 'HealthScore',
         'USER': 'root',
-        'PASSWORD': '', # change it with your sql server password for local testing
+        'PASSWORD': 'Wyx@0809ml', # change it with your sql server password for local testing
```
Comment (sarthakgoel1997) [inline healthScore/healthConfig/settings.py]: Please undo this change

File: healthScore/healthScore/migrations/__pycache__/0001_initial.cpython-39.pyc
Comment (sarthakgoel1997) [inline healthScore/healthScore/migrations/__pycache__/0001_initial.cpython-39.pyc:1]: It is a binary file, so we can skip pushing these

File: healthScore/healthScore/views.py [hunk 2]
```diff
@@ -27,8 +29,8 @@ def add_mock_data(request):
         hospital.objects.create(name="Hospital D", address="Address D", email="hospital_d@example.com", password="123456", contactInfo="123456781", status="pending")
 
         # Adding hospitalStaff data
-        # hospitalStaff.objects.create(hospitalID=hospital.objects.get(id=1), admin=True, name="Admin A", email="admin_a@hospitala.com", password="pass1234", specialization="", contactInfo="1234567890")
-        # hospitalStaff.objects.create(hospitalID=hospital.objects.get(id=1), admin=False, name="Doctor A", email="doctor_a@hospitala.com", password="pass1234", specialization="Anesthesiology", contactInfo="1234567890")
+        hospitalStaff.objects.create(hospitalID=hospital.objects.get(id=1), admin=True, name="Admin A", email="admin_a@hospitala.com", password="pass1234", specialization="", contactInfo="1234567890")
+        hospitalStaff.objects.create(hospitalID=hospital.objects.get(id=1), admin=False, name="Doctor A", email="doctor_a@hospitala.com", password="pass1234", specialization="Anesthesiology", contactInfo="1234567890")
```
Comment (sarthakgoel1997) [inline healthScore/healthScore/views.py]: Please comment these back so this dummy data is not added in production env

File: healthScore/healthScore/views.py [hunk 3]
```diff
@@ -37,10 +39,10 @@ def add_mock_data(request):
...
+        user.objects.create(email="user1@example.com", name="User1", password="userpass1@", userName="user1", dob="1990-01-01", contactInfo="1234567890", proofOfIdentity="Proof1", address="Address1", securityQues="", securityAns="",bloodGroup="A+")
+        user.objects.create(email="user2@example.com", name="User2", password="userpass2@", userName="user2", dob="1990-01-01", contactInfo="1234567890", proofOfIdentity="Proof2", address="Address2", securityQues="", securityAns="",bloodGroup="B+")
+        user.objects.create(email="user3@example.com", name="User3", password="userpass3@", userName="user3", dob="1990-01-01", contactInfo="1234567890", proofOfIdentity="Proof3", address="Address3", securityQues="", securityAns="",bloodGroup="O+")
+        user.objects.create(email="user4@example.com", name="User4", password="userpass4@", userName="user4", dob="1990-01-01", contactInfo="1234567890", proofOfIdentity="Proof4", address="Address4", securityQues="", securityAns="",bloodGroup="AB+")
```
Comment (sarthakgoel1997) [inline healthScore/healthScore/views.py]: Please comment these back so this dummy data is not added in production env

File: healthScore/requirements.txt [hunk 4]
```diff
@@ -11,6 +11,8 @@ Django==4.2
 filelock==3.13.1
 idna==3.6
 jmespath==1.0.1
+mysql-connector==2.2.9
+mysql-connector-python==8.3.0
```
Comment (sarthakgoel1997) [inline healthScore/requirements.txt]: Please remove these two. We don't need these - these requirements cause some deployment issues when we run `eb deploy`

File: healthScore/requirements.txt [hunk 5]
```diff
@@ -28,7 +30,7 @@ sqlparse==0.4.4
 termcolor==1.1.0
 typing_extensions==4.9.0
 tzdata==2024.1
-urllib3==1.26.18
+urllib3
```
Comment (sarthakgoel1997) [inline healthScore/requirements.txt]: Please restore the version before pushing so that no deployment issues occur later

File: healthScore/healthScore/views.py [hunk 6]
```diff
@@ -56,3 +58,28 @@ def add_mock_data(request):
...
+        gender = request.POST.get('gender')
+        street_address = request.POST.get('street_address')
+        city = request.POST.get('city')
+        state = request.POST.get('state')
+        phone_number = request.POST.get('phone_number')
+        identity_proof = request.POST.get('identity_proof')
+
+        if not user.objects.filter(email=email).exists():
+            user.objects.create(email=email, userName=username, password=password, name=fullname, 
+                                dob=dob, gender=gender, address=street_address, contactInfo=phone_number)
```
Comment (sarthakgoel1997) [inline healthScore/healthScore/views.py]: Here, we are storing the `address` as `street_address`. Let's concatenate the `city` and `state` fields as well, or take only one input from the user for address?

**Review by sarthakgoel1997 — CHANGES_REQUESTED** [review]

File: healthScore/healthConfig/settings.py (hunk 1 above)
Comment (jazwu) [reply healthScore/healthConfig/settings.py]: is it ok that you disapprove of this change when merging?

File: healthScore/healthScore/views.py (hunk 6 above)
Comment (jazwu) [reply healthScore/healthScore/views.py]: It is a problem. It was a workaround to run the app. Since the fields on the registration page don't exactly match the database, I was thinking we could modify both parts to make them consistent in the next sprint.

File: healthScore/healthConfig/settings.py (hunk 1 above)
Comment (sarthakgoel1997) [reply healthScore/healthConfig/settings.py]: Okay no issues, I will fix it

File: healthScore/healthScore/views.py (hunk 2 above)
Comment (jazwu) [reply healthScore/healthScore/views.py]: OK

File: healthScore/healthScore/views.py (hunk 6 above)
Comment (sarthakgoel1997) [reply healthScore/healthScore/views.py]: Ohh right. Is something like this possible?
`address = street_address + ', ' + city + ', ' + state`

**Comment by sarthakgoel1997** [comment]
Please take a pull of the develop branch and push it to resolve the conflicts

File: healthScore/.gitignore [hunk 7]
```diff
@@ -0,0 +1,2 @@
+**/__pycache__/*
+.DS_Store
```
Comment (shreyas2499) [inline healthScore/.gitignore]: This file need not be pushed

**Comment by jazwu** [comment]
> Please take a pull of the develop branch and push it to resolve the conflicts

merged!

File: healthScore/.gitignore (hunk 7 above)
Comment (jazwu) [reply healthScore/.gitignore]: Ok. But without it, cache files will be submitted every time I push to my remote repo. Is that ok?

Comment (shreyas2499) [reply healthScore/.gitignore]: You can keep the .git ignore file locally. But you won't need to push it

File: healthScore/healthConfig/settings.py [hunk 8]
```diff
@@ -81,14 +81,13 @@
     'default': {
         'ENGINE': 'django.db.backends.mysql',
         'NAME': 'HealthScore',
-        'USER': 'root',
-        'PASSWORD': '', # change it with your sql server password for local testing
-        'HOST':'localhost',
+        'USER': 'admin',
+        'PASSWORD': 'HealthScoreAdmin', # change it with your sql server password for local testing
+        'HOST':'awseb-e-meky4ftcfy-stack-awsebrdsdatabase-yqgueanvxaev.clkwykuegjia.us-west-2.rds.amazonaws.com',
```
Comment (sarthakgoel1997) [inline healthScore/healthConfig/settings.py]: Please restore the old config and avoid pushing prod DB credentials to GitHub

**Review by sarthakgoel1997 — CHANGES_REQUESTED** [review]

**Review by sarthakgoel1997 — APPROVED** [review]
